Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -0,0 +1,706 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import UIKit
+
+class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSource, UIPageViewControllerDelegate, MediaDetailViewControllerDelegate {
+
+    private struct MediaGalleryItem: Equatable {
+        let message: TSMessage
+        let attachmentStream: TSAttachmentStream
+        let viewController: MediaDetailViewController
+
+        var isVideo: Bool {
+            return attachmentStream.isVideo()
+        }
+
+        var image: UIImage {
```

Comment:
I wonder if this should be optional.  Missing attachment files are going to be more common once if/when we ship backup since we'll do "lazy restores" where attachments are filled in later.

Reply:
The point of this struct was to consolidate handling the optional, rather than sprinkling the handling of it throughout everything that touches this field. I'm hoping that's possible, as it makes reasoning elsewhere simpler.

Another approach which doesn't introduce nil would be to use a better placeholder image - e.g. in your example something that communicated "loading" would make sense to the user.